bigframes/bigquery/_operations/search.py

```diff
@@ -0,0 +1,243 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import json
+import typing
+from typing import Collection, Literal, Mapping, Optional, Union
+
+import bigframes_vendored.constants as constants
+
+import bigframes.core.sql
+import bigframes.ml.utils as utils
+
+if typing.TYPE_CHECKING:
+    import bigframes.dataframe as dataframe
+    import bigframes.series as series
+    import bigframes.session
+
+"""
+Search functions defined from
+https://cloud.google.com/bigquery/docs/reference/standard-sql/search_functions
+"""
+
+
+def create_vector_index(
+    table_id: str,
+    column_name: str,
+    *,
+    replace: bool = False,
+    index_name: Optional[str] = None,
+    distance_type="cosine",
+    stored_column_names: Collection[str] = (),
+    index_type: str = "ivf",
+    ivf_options: Optional[Mapping] = None,
+    tree_ah_options: Optional[Mapping] = None,
+    session: Optional[bigframes.session.Session] = None,
+) -> None:
+    """
+    Creates a new vector index on a column of a table.
+
+    This method calls the `CREATE VECTOR INDEX DDL statement
+    <https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#create_vector_index_statement>`_.
+
+    """
+    import bigframes.pandas
+
+    if index_name is None:
+        index_name = table_id.split(".")[-1]
```

Comment:
would reading into a TableReference be more robust?

Reply:
Pretty sure it's exactly the same logic there, but makes sense from the perspective of having one place for that logic (in case "." is allowed as a character in the table name someday).

Reply:
Done in latest commit (along with `identifier` logic).